pypose/module/lqr.py

```diff
@@ -158,41 +249,53 @@ def __init__(self, system, Q, p, T):
         assert self.Q.device == self.p.device, "device not compatible."
         assert self.Q.dtype == self.p.dtype, "tensor data type not compatible."
 
-    def forward(self, x_init):
+    def forward(self, x_init, current_x, current_u, time):
```

Comment:
Let's do not add these to the argument. These should be generated inside the class rather than let's the user provide. Because these are some intermediate variable you need to used for the convenience of the algorithm. Not necessary for using this class. See the original mpc.pytorch code.